cuml/random_forest_demo.ipynb

```diff
@@ -0,0 +1,286 @@
+    "\n",
+    "For information on converting your dataset to cuDF format, refer to the [cuDF documentation](https://rapidsai.github.io/projects/cudf/en/latest/)\n",
+    "\n",
+    "For additional information cuML's random forest model: https://rapidsai.github.io/projects/cuml/en/0.10.0/api.html#random-forest"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import cudf\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import pickle\n",
+    "\n",
+    "from cuml.ensemble import RandomForestClassifier as curfc\n",
+    "\n",
+    "from sklearn.ensemble import RandomForestClassifier as skrfc\n",
+    "from sklearn.metrics import accuracy_score\n",
+    "from sklearn.datasets import make_classification\n",
+    "from sklearn.model_selection import train_test_split"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define Parameters"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "n_samples = 2**17\n",
+    "n_features = 399\n",
+    "n_info = 300\n",
+    "data_type = np.float32"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Generate Data\n",
+    "\n",
+    "### Host"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "X,y = make_classification(n_samples=n_samples,\n",
+    "                          n_features=n_features,\n",
+    "                          n_informative=n_info,\n",
+    "                          random_state=123, n_classes=2)\n",
+    "\n",
+    "X = pd.DataFrame(X.astype(data_type))\n",
+    "y = pd.Series(y.astype(np.int32))\n",
+    "\n",
+    "X_train, X_test, y_train, y_test = train_test_split(X, y,\n",
+    "                                                    test_size = 0.2,\n",
+    "                                                    random_state=0)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### GPU"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "X_cudf_train = cudf.DataFrame.from_pandas(X_train)\n",
+    "X_cudf_test = cudf.DataFrame.from_pandas(X_test)\n",
+    "\n",
+    "y_cudf_train = cudf.Series(y_train.values)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Scikit-learn Model\n",
+    "\n",
+    "### Fit"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%time\n",
+    "sk_model = skrfc(n_estimators=40,\n",
+    "                 max_depth=16,\n",
+    "                 min_samples_split=2,\n",
```

Comment:
do we need `min_samples_split` to be specified here? Makes it deviate from the cuml example